proto/api.proto

```diff
@@ -0,0 +1,337 @@
+/* 
+ * Copyright 2021 Matrix Origin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+syntax = "proto3";
+package api;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "timestamp.proto";
+import "plan.proto";
+
+
+option go_package = "github.com/matrixorigin/matrixone/pkg/pb/api";
+option (gogoproto.sizer_all) = false;
+option (gogoproto.protosizer_all) = true;
+
+//types.type
```

Comment:
We have decided to not using typeid in proto.   Just use uint8 and use container/types.